share/eds/store.go

```diff
@@ -0,0 +1,211 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/filecoin-project/dagstore"
+	"github.com/filecoin-project/dagstore/index"
+	"github.com/filecoin-project/dagstore/mount"
+	"github.com/filecoin-project/dagstore/shard"
+	"github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+
+	"github.com/celestiaorg/celestia-node/share"
+
+	"github.com/celestiaorg/rsmt2d"
+)
+
+const (
+	blocksPath     = "/blocks/"
+	indexPath      = "/index/"
+	transientsPath = "/transients/"
+)
+
+type EDSStore struct { //nolint:revive
+	dgstr  *dagstore.DAGStore
+	bs     blockstore.Blockstore
+	mounts *mount.Registry
+
+	topIdx index.Inverted
+	carIdx index.FullIndexRepo
```

Comment:
they are not different for our purposes, they are only backreferences

Reply:
for reference: there was a discussion around whether these should exist on our EDSStore struct, since we can also access them through the `dgstr`, and they are not needed but should be kept for convenience for future use cases (IIRC @Wondertan 's argument)

Reply:
ok so we're keeping topIdx even though we don't really touch it?

Reply:
We do use it for testing at least, which counts

Reply:
@Wondertan im fine with either way, depends on what you both think. I don't think we need to store it on the struct